layers/eight_mile/utils.py

```diff
@@ -307,13 +307,12 @@ def iobes_mask(vocab, start, end, pad=None):
                         mask[vocab[to], vocab[from_]] = small
                     # Can only move to matching I or E
                     elif to.startswith("I-") or to.startswith("E-"):
-                        from_type = from_.split("-")[1]
-                        to_type = to.split("-")[1]
+                        from_type = from_.split("-", maxsplit=1)[1]
+                        to_type = to.split("-", maxsplit=1)[1]
                         if from_type != to_type:
                             mask[vocab[to], vocab[from_]] = small
-                elif from_.startswith(("E-", "I-", "S-", "O")):
+                elif from_.startswith(("E-", "S-", "O")):
                     # Can't move from E to I or E
```

Comment:
Why did we remove `I-` here? was that a bug before?

Reply:
It was superfluous, there is already a case in the if statements that handle `I-` so this being an `elif` meant it was never triggered.

Also if it was ever triggered it would have been a bug because you can't unilaterally say `I -> E` is illegal like the code here does, you need to check the types.